crates/model/src/solver_competition.rs

```diff
@@ -23,6 +23,18 @@ pub struct Request {
     pub transaction: Transaction,
     pub competition: SolverCompetitionDB,
     pub executions: Vec<(OrderUid, Execution)>,
+    pub scores: Scores,
+    pub participants: Vec<H160>, // solver addresses
```

Comment:
Instead of first adding the `auction_prices` when creating the auction in the autopilot, and then replacing them when we see a settlement, can't we just add the `auction_prices` here? This way we only ever add what is needed and don't need to have additional cleanup code for auction prices (as well as add new logic for cleaning up prices for auctions that didn't settle).

As it is, I think the `auction_prices` would cause a huge explosion of data in the database (as we never clean up `auction_prices` for auctions without settlements, which is most of them).

Reply:
> can't we just add the auction_prices here?

Yes, but didn't we want to move this responsibility to autopilot? 

I addressed the question with one possible solution here:
https://github.com/cowprotocol/services/pull/1271#issue-1599452805

Reply:
> Yes, but didn't we want to move this responsibility to autopilot?

Yes, but it will do it as part of the solver competition loop. Right now I have a very big concern that we never clean up this table and it will grow **very** quickly.

Since we already write a whole bunch of required data in the `solver::driver::run` when uploading the settlement competition, I really think we should move the external prices there as well.

Note that _eventually_ this will move to the `autopilot` here:

https://github.com/cowprotocol/services/blob/6d513aee1326b45365ebd1af7d290f1044423b2d/crates/autopilot/src/run_loop.rs#L71

For now though, I do think we are causing unneeded complication with the current setup.

Reply:
Alternatively we could add a cleanup task for auctions were no settlement attempt was made. Not sure what's better.

Reply:
We already write a bunch of required solver rewards data in the `solver` binary run loop (including things needed for this CIP like participants and scores). To me it makes sense to write it there (since we are already writing things) instead of splitting it into writing from two places. Also, in the `solver` run loop, we already have all the required information to write exactly the `auction_price`s we need.

Reply:
fixed here https://github.com/cowprotocol/services/pull/1282